core/src/main/java/org/stellar/anchor/sep31/Sep31DepositInfoGenerator.java

```diff
@@ -0,0 +1,13 @@
+package org.stellar.anchor.sep31;
+
+import org.stellar.anchor.api.sep.sep31.Sep31DepositInfo;
+
+public interface Sep31DepositInfoGenerator {
```

Comment:
If the deposit info is specific to `platform`, we can move this class to `platform` subproject. But I start to see that you plan to make the deposit info generation customizable. 👍

Reply:
Alright, this is what I ended up doing:
* The `Sep31DepositInfo` class and the `Sep31DepositInfoGenerator` interface stayed in core because they are used for the Sep31Service.
* I moved the `Sep31DepositInfoGeneratorSelf` implementation to platform.
* I created a `Sep31DepositInfoGeneratorCircle` implementation class, also in platform.
* `payments` doesn't hold any reference to `Sep31DepositInfo` nor `Sep31DepositInfoGenerator` anymore.

Reply:
I think this is a good interim solution. 

After MVP, What I am thinking is to implement filter or observer interface pattern to generalize the interface and reduce the need for modifications when we have more use cases. 

I will come up with a simple architecture design and discuss with you.